### ISSUE TITLE
learn.show_results() output no longer returns original image 

### ISSUE THREAD

**Issue by WRFitch** [fastai/fastai#3234]
**Only file a bug report here when you're quite confident it's not an issue with your local setup.**
This error has only appeared since the new update, and reverting code to a version committed before the update has no impact on the error.

Please confirm you have the latest versions of fastai, fastcore, and nbdev prior to reporting a bug (delete one): YES (I haven't got nbdev but I never needed it before version 2.2.7 - if it's necessary to use fastai, could that be written in the docs?) 

**Describe the bug**
When viewing model output in a colab notebook using `learn.show_results()`, the image returned is very low-contrast, and tinted beige. This can be removed by passing the original DataLoader to `show_results` as the `dl` parameter, but this is not documented in the new release. 

**To Reproduce**
1. Set up a colab notebook and update fastai to the latest version 
2. Define a DataBlock thusly:
```
test_block = DataBlock(
    blocks = (ImageBlock, RegressionBlock),
    get_items = getImgs,
    get_y = getData,
    item_tfms = Resize(224)
)
test_dl = test_block.dataloaders(f{"a directory containing some PNG images}")
```
3. Show a batch from the dataloader, to verify the images are ok going in:
`test_dl.show_batch(nrows=9, max_n=9, figsize = (50,50))`
4. Define a cnn_learner with the given dataloader (resnet depth doesn't seem to matter, but I'm using 152):
`learn = cnn_learner(test_dl, resnet152, y_range=(0, 100),  metrics=rmse)`
5. Display results from the learner, the issue exists before and after training: 
`learn.show_results(ds_idx=1, nrows=9, max_n=9, figsize = (50,50))`
6. To fix, include dataloader in show_results:
`learn.show_results(ds_idx=1, dl=test_dl, nrows=9, max_n=9, figsize = (50,50))`

**Expected behavior**
learn.show_results should show no distortion, in the same way as the dataloader `show_batch` method. 

**Error with full stack trace**

No stacktrace since no error was reported

**Additional context**
Environment setup is standard Colab setup, with a GPU runtime. 

Intended image (result of stage 6 in example):
![image](https://user-images.githubusercontent.com/36193438/108883536-75e43d80-75fd-11eb-88f2-6c2d7ab733e8.png)

Actual image (result of stage 5 in example): 
![image](https://user-images.githubusercontent.com/36193438/108883623-96ac9300-75fd-11eb-93d5-7d22218f94da.png)

**Comment by Ankur-singh** [comment]
Yes, I am also having the same issue! Look at the below image. All the leafs are originally green. 

![image](https://user-images.githubusercontent.com/14893062/111725997-10970b80-888e-11eb-912f-0dcda2718c25.png)

**Comment by BobMcDear** [comment]
I'm having the same issue on multiple machines. 

A couple of notes:

1. The issue isn't with ```Learner.show_results```; it is with ```Learner.predict```.
2. It can be fixed by reinitializing (recreating) the DataLoaders object after creating the Learner and reassigning it (```Learner.dls = new_dls```).

**Comment by BobMcDear** [comment]
I suspect this is related to a [bug](https://github.com/fastai/fastai/issues/3281) where fastai only normalizes the training set.

**Comment by yang20085936** [comment]
I have the same question and it can be fixed as @BobMcDear  said. Thanks!

> A couple of notes:
> 
>     1. It can be fixed by reinitializing (recreating) the DataLoaders object after creating the Learner and reassigning it (`Learner.dls = new_dls`).
> 
>     2. This only happens for the validation set, not the training set.

**Comment by BobMcDear** [comment]
@yang20085936 Please take a look at [this](https://github.com/fastai/fastai/issues/3281). The correct way to fix it (for now) is to normalize your data through your DataBlock (```batch_tfms=Normalize.from_stats(*imagenet_stats)```) rather than the Learner. 

The method I proposed is flawed given that it causes your data not to get normalized if you're DataBlock isn't normalizing the data.

**Comment by BobMcDear** [comment]
For anyone having this issue, please make sure you've installed the latest version of fastai via Github, not pip. [This](https://github.com/fastai/fastai/issues/3281), which also causes this bug, is only present in the pip version of fastai.

**Comment by yang20085936** [comment]
I followed your new method and it works very well!  Thanks again! @BobMcDear

**Comment by BobMcDear** [comment]
@yang20085936 No problem!
